code/game/objects/items.dm

```diff
@@ -99,6 +99,8 @@
 	can_block_movement = FALSE
 
 /obj/item/New()
+	atom_flags |= ATOM_FLAG_UNTARGETABLE //items on floor shouldn't be a target for guns
```

Comment:
Flags are set at authortime, not in New
Set them in the block above

Figure out which other flags items have by default and append this one

Doing it in New is not good because this code will prevent the possibility of making targetable items, which is very much needed. For example, computers are all items, even the dense anchored consoles. Make sure to set the atom flags on those

Reply:
> Figure out which other flags items have by default and append this one
Sorry, but I don't have that much time to check every item in the game for atom_flags just to put them in.

> Doing it in New is not good because this code will prevent the possibility of making targetable items, which is very much needed. For example, computers are all items, even the dense anchored consoles. Make sure to set the atom flags on those

It can be done just by this:
```
/obj/item/New()
   . = ..()
  atom_flags &= ~ATOM_FLAG_UNTARGETABL
```
Some items override atom_flags, which will cause deleting of the flag. So I did it to ensure, that every item in subtype will have it (unless removed with a new New).
Implecation with the flag just in main object:
![xYwn8981Vi](https://user-images.githubusercontent.com/19491666/103782844-8e5ac180-506a-11eb-9774-4d969ceeb66d.gif)
Vials override it.

Reply:
this is not how things are done, having code in a base class to set or unset things gets messy, it creates traps for the unaware coder, and it will be hell to track down in future. It is generally a violation of coding standards.

You dont have to check every item in the game, just the base item class and set it there. If various things are setting their own flags and wouldn't be affected by this change, so be it. those problems can be dealt with in future, when and if they arise.

Trying to pre-emptively solve them with this approach is going to create worse problems in the long run